pkg/controller/provider/web/client.go

```diff
@@ -167,25 +133,9 @@ func (r *ProviderClient) List(resource interface{}, param ...Param) (err error)
 //   ProviderNotReadyErr
 //   NotFoundErr
 func (r *ProviderClient) Watch(resource interface{}, h EventHandler) (w *Watch, err error) {
-	err = r.find()
```

Comment:
Don't attempt to fetch the provider resource first and examine the X-Provider header instaed.
Smarter and reduces API requests.